lib/martian_robot.rb

```diff
@@ -1,31 +1,49 @@
+require_relative 'movements'
+require_relative 'mars'
+
 class MartianRobot
 
   attr_reader :coordinates, :orientation
 
-  def initialize(coordinates, orientation)
+  def initialize(coordinates, orientation, planet)
     @coordinates = coordinates
     @orientation = orientation
     @movements = Movements.new
+    @planet = planet
   end
 
   def report_final_position(instruction)
     final_position = starting_position
-    instruction.split('').each do |single_instruction|
-      if single_instruction == 'F'
-        @movements.move_forward(@orientation, final_position)
-      elsif single_instruction == 'R'
-        @orientation = @movements.change_orientation_clockwise(@orientation)
-      elsif single_instruction == 'L'
-        @orientation = @movements.change_orientation_anticlockwise(@orientation)
-      end
-    end
-    final_position
+    instruction.split('').each {|single_instruction| execute(single_instruction, final_position)}
+    is_robot_lost?(final_position) ? :lost : final_position
   end
 
+
   def starting_position
     x = @coordinates[0]
     y = @coordinates[1]
     [x,y]
   end
 
+  private
+
+  def execute(instruction, position)
+    case instruction
+    when'F'
+      @movements.move_forward(@orientation, position)
+    when 'R'
+      @orientation = @movements.change_orientation_clockwise(@orientation)
+    when 'L'
+      @orientation = @movements.change_orientation_anticlockwise(@orientation)
+    end
+  end
+
+  def is_robot_lost?(final_position)
```

Comment:
It's a Ruby naming convention to not use "is" in the method name of a boolean method. In this case you could just call it `lost?` Because the method is in the class MartianRobot, we know that it will be the robot who is lost.